cli/cmd/uninstall/uninstall_test.go

```diff
@@ -414,8 +434,165 @@ func TestTaskCreateCommand_AutocompleteArgs(t *testing.T) {
 	assert.Equal(t, complete.PredictNothing, c)
 }
 
+func TestFetchCustomResources(t *testing.T) {
+	cr := unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "consul.hashicorp.com/v1alpha1",
+			"kind":       "ServiceDefaults",
+			"metadata": map[string]interface{}{
+				"name":      "server",
+				"namespace": "default",
+			},
+		},
+	}
+	nonConsulCR := unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "example.com/v1",
+			"kind":       "Example",
+			"metadata": map[string]interface{}{
+				"name":      "example-resource",
+				"namespace": "default",
+			},
+		},
+	}
+
+	c := getInitializedCommand(t, nil)
+	c.apiext, c.dynamic = createClientsWithCrds()
+
+	_, err := c.dynamic.Resource(serviceDefaultsGRV).Namespace("default").Create(context.Background(), &cr, metav1.CreateOptions{})
+	require.NoError(t, err)
+	_, err = c.dynamic.Resource(nonConsulGRV).Namespace("default").Create(context.Background(), &nonConsulCR, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	crds, err := c.fetchCustomResourceDefinitions()
+	require.NoError(t, err)
+
+	actual, err := c.fetchCustomResources(crds)
+	require.NoError(t, err)
+	require.Len(t, actual, 1)
+	require.Contains(t, actual, cr)
+	require.NotContains(t, actual, nonConsulCR)
```

Comment:
👏

Reply:
+1 great job on test! It'd be nice to add another CR in a different k8s namespace but not a blocker.

Reply:
Thank you! I added the CR in another namespace